apps/dgst.c

```diff
@@ -413,6 +413,34 @@ int dgst_main(int argc, char **argv)
     return ret;
 }
 
+static const char *newline_escape_filename(const char *file, int *backslash)
+{
+    int i = 0, length = strlen(file), newline_count = 0;
+    while(i < length) {
+        const char c = file[i];
+        if (c == '\n') {
+            newline_count++;
+        }
+        i++;
+    }
+    char *file_cpy = app_malloc(length + newline_count, file);
+    i = 0;
+
+    while(i < length) {
+        const char c = file[i];
```

Comment:
This misses the character after the newline.  You'll need two counters, one to index into file and the other for file_cpy.

Reply:
Great catch.  Coming in my new commit.